src/sidebar/services/router.js

```diff
@@ -49,8 +59,11 @@ export default function router($window, store) {
 
   /**
    * Generate a URL for a given route.
+   *
+   * @param {string} name
+   * @param {Object.<string,string>} params
    */
-  function routeUrl(name, params = {}) {
+  routeUrl(name, params = {}) {
     let url;
     const queryParams = { ...params };
```

Comment:
```suggestion
    const {id, queryParams} =  params;
```
And then 3 lines (73-75) can be deleted

Reply:
On this line `id` may or may not be a URL parameter for the route. Its only within the switch branches that we know what the valid route params are.

I expect in future as our routing needs get more sophisticated we'll introduce a better abstraction or possibly re-use some library to handles the route name <-> URL conversion process.

Reply:
If the other two routes don't use `id`  then it is safe.